src/ZeroLog/JsonWriter.cs

```diff
@@ -0,0 +1,319 @@
+                    stringBuffer.Append('"');
+
+                    dataPointer += sizeof(char);
+                    break;
+
+                case ArgumentType.Int16:
+                    stringBuffer.Append(*(short*)dataPointer, StringView.Empty);
+                    dataPointer += sizeof(short);
+                    break;
+
+                case ArgumentType.Int32:
+                    stringBuffer.Append(*(int*)dataPointer, StringView.Empty);
+                    dataPointer += sizeof(int);
+                    break;
+
+                case ArgumentType.Int64:
+                    stringBuffer.Append(*(long*)dataPointer, StringView.Empty);
+                    dataPointer += sizeof(long);
+                    break;
+
+                case ArgumentType.Single:
+                    stringBuffer.Append(*(float*)dataPointer, StringView.Empty);
+                    dataPointer += sizeof(float);
+                    break;
+
+                case ArgumentType.Double:
+                    stringBuffer.Append(*(double*)dataPointer, StringView.Empty);
+                    dataPointer += sizeof(double);
+                    break;
+
+                case ArgumentType.Decimal:
+                    stringBuffer.Append(*(decimal*)dataPointer, StringView.Empty);
+                    dataPointer += sizeof(decimal);
+                    break;
+
+                case ArgumentType.Guid:
+                    stringBuffer.Append('"');
+                    stringBuffer.Append(*(Guid*)dataPointer, StringView.Empty);
+                    stringBuffer.Append('"');
+                    dataPointer += sizeof(Guid);
+                    break;
+
+                case ArgumentType.DateTime:
+                    stringBuffer.Append('"');
+                    stringBuffer.Append(*(DateTime*)dataPointer, StringView.Empty);
+                    stringBuffer.Append('"');
+                    dataPointer += sizeof(DateTime);
+                    break;
+
+                case ArgumentType.TimeSpan:
+                    stringBuffer.Append('"');
+                    stringBuffer.Append(*(TimeSpan*)dataPointer, StringView.Empty);
+                    stringBuffer.Append('"');
+                    dataPointer += sizeof(TimeSpan);
+                    break;
+
+                case ArgumentType.Enum:
+                    var enumArg = (EnumArg*)dataPointer;
+                    dataPointer += sizeof(EnumArg);
+                    enumArg->AppendTo(stringBuffer);
+                    break;
+
+                case ArgumentType.Null:
+                    stringBuffer.Append("null");
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private static bool NeedsEscaping(string s)
+        {
+            // Not using linq here since it (might) add overhead.
+            foreach (var c in s)
+            {
+                if (c < '\u001F' || c == '"' || c == '/')
```

Comment:
You probably meant `\` instead of `/`, but this method should not be needed anyway (see comment above)